File: CM_U3D_Dev/LuaProject/Core/Util/TextUtil.lua
--文本工具
TextUtil = { }

--字符串转化为整形数组
function TextUtil.StrToIntArr(str, splitStr)
    if str == nil or str == "" then
        return nil;
    end
    if splitStr == nil then
        splitStr = ',';
    end

    local paramStrs = string.split(str, splitStr);
    local value = {}
    for i = 1, #paramStrs do
        value[i] = tonumber(paramStrs[i]);
    end
    return value;
end

--字符串转化为2维整形数组
function TextUtil.StrToIntArr2(str, splitStr1, splitStr2)
    if str == nil or str == "" then
        return nil;
    end
    if splitStr1 == nil then
        splitStr1 = '|';
    end
    if splitStr2 == nil then
        splitStr2 = ',';
    end

    local paramStrs = string.split(str, splitStr1);
    local value = {};
    for i = 1, #paramStrs do
        value[i] = TextUtil.StrToIntArr(paramStrs[i], splitStr2);
    end
    return value;
end

function TextUtil.StrToFloatArr(str, splitStr)
    if str == nil or str == "" then
        return nil;
    end
    if splitStr == nil then
        splitStr = ',';
    end
    local paramStrs = string.split(str, splitStr);
    local value = {};
    for i = 1, #paramStrs do
        value[i] = tonumber(paramStrs[i]);
    end
    return value;
end

function TextUtil.StrToFloatArr2(str, splitStr1, splitStr2)
    if str == nil or str == "" then
        return nil;
    end
    if splitStr1 == nil then
        splitStr1 = '|';
    end
    if splitStr2 == nil then
        splitStr2 = ',';
    end
    local paramStrs = string.split(str, splitStr1);
    local value = {};
    for i = 1, #paramStrs do
        value[i] = TextUtil.StrToFloatArr(paramStrs[i], splitStr2);
    end
    return value;
end

--将(秒)转换成时分秒 格式：00:00:00
function TextUtil.GetTimeStr(time, isHour)
    if isHour == nil then
        isHour = false;
    end
    if time > 0 then
        local s = math.modf(time % 60); --剩余秒数
        local temp = math.modf(time / 60)
        local m = temp % 60;    --剩余分钟
        temp = math.modf(time / 60 / 60)
        local h = temp % 24;    --剩余小时
        if h > 0 or isHour then
            return string.format("%.2d:%.2d:%.2d", h, m, s);
        else
            return string.format("%.2d:%.2d", m, s);
        end
    else
        if isHour then
            return "00:00:00";
        else
            return "00:00";
        end
    end
end

--将(秒)转换成时分秒 格式：00:00:00
function TextUtil.GetMinuteStr(time)
    if time > 0 then
        local s = math.modf(time % 60); --剩余秒数
        local m = math.modf(time / 60)
        return string.format("%.2d:%.2d", m, s);
    else
        return "00:00";
    end
end

--获取时间的天时分秒
function TextUtil.TimeToDHMSTb(time)
    local day = math.floor(time / 86400)
    local hour = math.floor((time % 86400) / 3600)
    local min = math.floor((time % 3600) / 60)
    local sec = time % 60
    return {day = day, hour = hour , min = min, sec = sec}
end

--时分秒转时间
function TextUtil.HMSToTime(hour, min, sec)
    return TextUtil.DHMSToTime(0, hour, min, sec)
end

--天时分秒转时间
function TextUtil.DHMSToTime(day, hour, min, sec)
    return day * 86400 + hour * 3600 + min * 60 + sec
end

function TextUtil.ToBase64String(bytes)
    return CS.System.Convert.ToBase64String(bytes);
end

function TextUtil.FromBase64String(string)
    return CS.System.Convert.FromBase64String(string)
end

return TextUtil

File: CM_U3D_Dev/Assets/Scripts/Util/TextUtil.cs
using System.Collections;
using System.Collections.Generic;
using System;
using UnityEngine;

/// <summary>
/// DEV: 张柏泉
/// 文本工具类
/// </summary>
public class TextUtil  {
    //字符串转化为整形数组
    public static int[] StrToIntArr(string str, char splitStr = ',') {
        if (str == null || str == "") {
            return null;
        }
        string[] paramStrs = str.Split(splitStr);
        int[] value = new int[paramStrs.Length];
        for (int i = 0; i < value.Length; i++) {
            value[i] = Convert.ToInt32(paramStrs[i]);
        }
        return value;
    }

    //字符串转化为2维整形数组
    public static int[][] StrToIntArr2(string str, char splitStr1 = '|', char splitStr2 = ',') {
        if (str == null || str == "") {
            return null;
        }
        string[] paramStrs = str.Split(splitStr1);
        int[][] value = new int[paramStrs.Length][];
        for (int i = 0; i < value.Length; i++) {
            value[i] = StrToIntArr(paramStrs[i], splitStr2);
        }
        return value;
    }

    public static float[] StrToFloatArr(string str, char splitStr = ',') {
        if (str == null || str == "") {
            return null;
        }
        string[] paramStrs = str.Split(splitStr);
        float[] value = new float[paramStrs.Length];
        for (int i = 0; i < value.Length; i++) {
            value[i] = Convert.ToSingle(paramStrs[i]);
        }
        return value;
    }

    public static float[][] StrToFloatArr2(string str, char splitStr1 = '|', char splitStr2 = ',') {
        if (str == null || str == "") {
            return null;
        }
        string[] paramStrs = str.Split(splitStr1);
        float[][] value = new float[paramStrs.Length][];
        for (int i = 0; i < value.Length; i++) {
            value[i] = StrToFloatArr(paramStrs[i], splitStr2);
        }
        return value;
    }

    /// <summary>
    /// 将(秒)转换成时分秒 格式：00:00:00
    /// </summary>
    /// <param name="time"></param>
    /// <returns></returns>
    public static string GetTimeStr(int time, bool isHour=false) {
        if (time > 0) {
            int s = (int)(time % 60);// 剩余秒数
            int m = (int)(time / 60) % 60;// 剩余分钟
            int h = (int)(time / 60 / 60) % 24;// 剩余小时
            if (h > 0 || isHour) {
                return string.Format("{0:D2}:{1:D2}:{2:D2}", h, m, s);
            } else {
                return string.Format("{0:D2}:{1:D2}", m, s);
            }
        } else {
            if (isHour) {
                return "00:00:00";
            } else {
                return "00:00";
            }
        }
    }

    /// <summary>
    /// 字符串转float
    /// </summary>
    /// <param name="str">字符串</param>
    /// <returns></returns>
    public static float StrToFloat(string str) {
        //if (str == null || str.Equals(""))
        if (str == null || str.Length <= 0) {
            return 0;
        }
        float returnValue = 0;
        try {
            if (str.StartsWith("0x") || str.StartsWith("0X")) {
                returnValue = (float)Convert.ToDouble(str.Substring(2));
            } else
                if (str.IndexOf(".") != -1) {
                returnValue = (float)Convert.ToDouble(str);
            } else {
                returnValue = (float)Convert.ToDouble(str);
            }
        } catch (Exception ex) {
        }
        return returnValue;
    }
}
